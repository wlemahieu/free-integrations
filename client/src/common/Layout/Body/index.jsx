import React from 'react';
import PropTypes from 'prop-types';
import { Switch, Route, withRouter } from 'react-router-dom';
import { Layout } from 'antd';
import BreadcrumbTrail from 'common/Layout/BreadcrumbTrail';
import Home from 'common/Routes/Home';
import Chat from 'common/Routes/Chat';
import License from 'common/Routes/License';
import News from 'common/Routes/News';

const { Content } = Layout;
const Body = props => {
  const { location } = props;
  return (
    <Content className="content">
      <BreadcrumbTrail location={location} />
      <Switch>
        <Route path="/news">
          <News />
        </Route>
        <Route path="/license">
          <License />
        </Route>
        <Route path="/chat">
          <Chat />
        </Route>
        <Route path="/">
          <Home />
        </Route>
      </Switch>
    </Content>
  );
};

Body.propTypes = {
  location: PropTypes.object
};

Body.defaultProps = {
  location: {}
};

export default withRouter(Body);
