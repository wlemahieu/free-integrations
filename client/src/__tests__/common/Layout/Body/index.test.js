import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Body from 'common/Layout/Body';

describe('Layout Body', () => {
  it('should render', () => {
    const { container } = render(
      <Router>
        <Body location={{}} />;
      </Router>
    );
    expect(container).toMatchSnapshot();
  });
});
