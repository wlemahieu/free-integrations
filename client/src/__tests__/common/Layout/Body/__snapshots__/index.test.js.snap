// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Body should render 1`] = `
<div>
  <main
    class="content ant-layout-content"
  >
    <div
      class="ant-breadcrumb"
    >
      <span>
        <span
          class="ant-breadcrumb-link"
        >
          <a
            href="/home"
            title="Home"
          >
            Home
          </a>
        </span>
        <span
          class="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </div>
    <div
      class="ant-row centered"
    >
      <div
        class="ant-col ant-col-12 ant-col-offset-6"
      >
        <h1>
          Greetings!
        </h1>
      </div>
    </div>
  </main>
  ;
</div>
`;
